### DISCUSSION TITLE
KubernetesPodOperator Mark Task Failed after SUCCESS Randomly

### DISCUSSION THREAD

**Discussion by armandleopold** [apache/airflow#21558] [General]
### Apache Airflow version

2.2.3 (latest released)

### What happened

We are using KubernetesPodOperator to schedule basically all of our tasks.
Sometimes, randomly some TASK finish and are marked as SUCCESS and then immediately marked FAILED.

```
....
2022-02-08, 06:09:51 CET - Bootstrap - INFO - *********************************************************
2022-02-08, 06:09:51 CET - Bootstrap - INFO - ***                        FIN                        ***
2022-02-08, 06:09:51 CET - Bootstrap - INFO - *********************************************************
Event: consore-biologieclinique-interface-0.4.f3afe897abf646c6b22642cb594b8745 had an event of type Succeeded
Event with job id consore-biologieclinique-interface-0.4.f3afe897abf646c6b22642cb594b8745 Succeeded
Event: consore-biologieclinique-interface-0.4.f3afe897abf646c6b22642cb594b8745 had an event of type Succeeded
Event with job id consore-biologieclinique-interface-0.4.f3afe897abf646c6b22642cb594b8745 Succeeded
Marking task as SUCCESS. dag_id=consore-by-window, task_id=consore-biologieclinique-interface-0.4, execution_date=20220207T215900, start_date=20220208T040444, end_date=20220208T040953
Task exited with return code 0
Marking task as FAILED. dag_id=consore-by-window, task_id=consore-biologieclinique-interface-0.4, execution_date=20220207T215900, start_date=20220208T040444, end_date=20220208T040953
/opt/bitnami/airflow/venv/lib/python3.8/site-packages/airflow/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_start' or 'logical_date' instead.
/opt/bitnami/airflow/venv/lib/python3.8/site-packages/airflow/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'next_ds' from the template is deprecated and will be removed in a future version. Please use '{{ data_interval_end | ds }}' instead.
/opt/bitnami/airflow/venv/lib/python3.8/site-packages/airflow/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'next_ds_nodash' from the template is deprecated and will be removed in a future version. Please use '{{ data_interval_end | ds_nodash }}' instead.
/opt/bitnami/airflow/venv/lib/python3.8/site-packages/airflow/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'next_execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_end' instead.
/opt/bitnami/airflow/venv/lib/python3.8/site-packages/airflow/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'prev_ds' from the template is deprecated and will be removed in a future version.
/opt/bitnami/airflow/venv/lib/python3.8/site-packages/airflow/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'prev_ds_nodash' from the template is deprecated and will be removed in a future version.
/opt/bitnami/airflow/venv/lib/python3.8/site-packages/airflow/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'prev_execution_date' from the template is deprecated and will be removed in a future version.
/opt/bitnami/airflow/venv/lib/python3.8/site-packages/airflow/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'prev_execution_date_success' from the template is deprecated and will be removed in a future version. Please use 'prev_data_interval_start_success' instead.
/opt/bitnami/airflow/venv/lib/python3.8/site-packages/airflow/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'tomorrow_ds' from the template is deprecated and will be removed in a future version.
/opt/bitnami/airflow/venv/lib/python3.8/site-packages/airflow/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'tomorrow_ds_nodash' from the template is deprecated and will be removed in a future version.
/opt/bitnami/airflow/venv/lib/python3.8/site-packages/airflow/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'yesterday_ds' from the template is deprecated and will be removed in a future version.
/opt/bitnami/airflow/venv/lib/python3.8/site-packages/airflow/utils/context.py:152 AirflowContextDeprecationWarning: Accessing 'yesterday_ds_nodash' from the template is deprecated and will be removed in a future version.
/opt/bitnami/airflow/venv/lib/python3.8/site-packages/airflow/utils/email.py:102 PendingDeprecationWarning: Fetching SMTP credentials from configuration variables will be deprecated in a future release. Please set credentials using a connection instead.
section/key [smtp/smtp_user] not found in config
Email alerting: attempt 1
...
```

### What you expected to happen

What i expect is that the TASK is Marked as SUCCESS only .

```
...
2022-02-08, 06:09:51 CET - Bootstrap - INFO - *********************************************************
2022-02-08, 06:09:51 CET - Bootstrap - INFO - ***                        FIN                        ***
2022-02-08, 06:09:51 CET - Bootstrap - INFO - *********************************************************
Event: consore-biologieclinique-interface-0.4.f3afe897abf646c6b22642cb594b8745 had an event of type Succeeded
Event with job id consore-biologieclinique-interface-0.4.f3afe897abf646c6b22642cb594b8745 Succeeded
Event: consore-biologieclinique-interface-0.4.f3afe897abf646c6b22642cb594b8745 had an event of type Succeeded
Event with job id consore-biologieclinique-interface-0.4.f3afe897abf646c6b22642cb594b8745 Succeeded
Marking task as SUCCESS. dag_id=consore-by-window, task_id=consore-biologieclinique-interface-0.4, execution_date=20220207T215900, start_date=20220208T040444, end_date=20220208T040953
Task exited with return code 0
...
```

### How to reproduce

I don't know how to reproduce.

Ideas : 

* Deploy airflow 2.2.3
* Create DAG with KubernetesPodOperator
* Schedule a task

### Operating System

debian-buster

### Versions of Apache Airflow Providers

can't have that

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

using the bitnami helm https://artifacthub.io/packages/helm/bitnami/airflow

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by raphaelauv** [comment]
what version of https://pypi.org/project/apache-airflow-providers-cncf-kubernetes/#history ?

**Comment by armandleopold** [comment]
@raphaelauv  I am not really familiar with how bitnami builds its images : 
https://github.com/bitnami/bitnami-docker-airflow/blob/master/2/debian-10/prebuildfs/opt/bitnami/.bitnami_components.json#L7

**Comment by raphaelauv** [comment]
you can go inside the container and do `pip list`

**Comment by armandleopold** [comment]
```
root@test-6494ccc5b8-7nlf6:/home# python -m pip list
Package      Version
------------ -------
distlib      0.3.4
filelock     3.4.2
pip          21.3.1
platformdirs 2.4.1
setuptools   46.4.0
six          1.16.0
virtualenv   20.11.0
```

**Comment by armandleopold** [comment]
This is strange : 
The task is "Failed" but don't appears as failed in the front-end DAG view
![image](https://user-images.githubusercontent.com/7100281/153830351-15f5e9f0-3555-4a0c-af95-c67a3d9f37c1.png)
![image](https://user-images.githubusercontent.com/7100281/153830421-13d908a9-09dc-4d2b-a69a-fa3ec52b45b8.png)
![image](https://user-images.githubusercontent.com/7100281/153830482-73e33929-0187-4767-97cc-f6b1e8d01cb9.png)

Ho no ok, it's because the last DAG run finished Successfully.

![image](https://user-images.githubusercontent.com/7100281/153830820-477f0e65-db06-41d7-827c-db1f53997dbe.png)

**Comment by potiuk** [comment]
Converted into an issue until we have clear reproduction path and we know this is an airflow issue.

You can find list of providers in 2.2.3 in admin part of the UI or by running "airflow providers" CLI command.

From the logs I believe you have a resource isue in your deployment and Kubernetes simply kills the pods randomly and does not give them a chance to report back (for example because they do not have enough memory or for other reasons). Please investigate your Kubernetes Logs and  increase the resources that are needed in your deployment.

**Reply by armandleopold** [reply]
Thanks @potiuk , will look into that !

@raphaelauv 

<html><body>


apache-airflow-providers-cncf-kubernetes | 2.2.0 | Kubernetes
-- | -- | --



</body>
</html>

**Reply by raphaelauv** [reply]
@armandleopold update to the latest version 3.0.2

**Reply by potiuk** [reply]
Yeah - if you can't find a clue in yuor K8S logs upgrading as suggested by @raphaelauv might be good idea. 

 There were quite many changes (but also some breaking changes) since: https://airflow.apache.org/docs/apache-airflow-providers-cncf-kubernetes/stable/index.html including upgrading to newer versions of K8S libraries.

**Reply by potiuk** [reply]
And if you look for how to upgrade - I am not sure about Bitnami image - because that is not our "official" image but you can find some upgrade scenarios in our "official" image documentation that you might find useful (or you might want to switch to the official image): 

https://airflow.apache.org/docs/docker-stack/build.html#examples-of-image-extending

**Reply by armandleopold** [reply]
I think we are going to switch to your official helm chart yes.
